### PR TITLE
EXLM-2450 Recommended block - selecting all products

### PR DESCRIPTION
**Pull request by ruhisingh1**
Please always provide the Jira Issue your PR is for, as well as test URLs where your change can be observed (before and after):

Jira ID: EXLM-2450

Test URLs:

- Before: https://main--exlm--adobe-experience-league.hlx.live?martech=off
- After: https://exlm-2450--exlm--adobe-experience-league.hlx.live?martech=off

### CODE REVIEW

**Review by gauthamksn — APPROVED** [review]

File: blocks/recommended-content/recommended-content.js [hunk 1]
```diff
@@ -194,7 +194,7 @@ export default async function decorate(block) {
   const headerContainer = block.querySelector('.recommended-content-header');
   const descriptionContainer = block.querySelector('.recommended-content-description');
   const reversedDomElements = remainingElements.reverse();
-  const [firstEl, secondEl, thirdEl, fourthEl, fifthEl, ...otherEl] = reversedDomElements;
+  const [linkEl, resultTextEl, sortEl, optionsEl, solutionEl, ...contentTypesEl] = reversedDomElements;
```
Comment (ruhisingh1) [inline blocks/recommended-content/recommended-content.js:197]: One of the fields (either roles or filterProductBy) is missing here.

File: blocks/recommended-content/recommended-content.js [hunk 2]
```diff
@@ -405,9 +405,10 @@ export default async function decorate(block) {
         return expLevel;
       });
       const sortCriteria = COVEO_SORT_OPTIONS[sortByContent?.toUpperCase() ?? 'MOST_POPULAR'];
-      const role = fourthEl?.innerText?.trim()?.includes('profile_context')
+      const filterProductByOption = optionsEl?.innerText?.trim() ?? '';
+      const role = filterProductByOption?.includes('profile_context')
```
Comment (ruhisingh1) [inline blocks/recommended-content/recommended-content.js:409]: Role is not dependent on filterProductBy field data.

File: blocks/recommended-content/recommended-content.js (hunk 1 above)
Comment (nitin-rachabathuni) [reply blocks/recommended-content/recommended-content.js]: done Ruhi

File: blocks/recommended-content/recommended-content.js (hunk 2 above)
Comment (nitin-rachabathuni) [reply blocks/recommended-content/recommended-content.js]: corrected Ruhi

**Review by nitin-rachabathuni — APPROVED** [review]

**Review by nitin-rachabathuni — APPROVED** [review]

**Pull request merged**